### PR TITLE
Custom ttl options for each collection and fixed tests

### PR DESCRIPTION
**Pull request by robertsLando**
@mcollina 

### Changes

- **Feat**: option to set custom TTL to each collection: subscriptions, incoming, outgoing, will, retained (keep back compatibility)
- **Feat**: Set ttl to -1 to make a collection persistent (no ttl to docs)
- **Fix**: Remove previous created indexes on connect if `dropExistingIndexes` option is specified
- Added test for `dropExistingIndexes`
- **Fix**: deprecation warning of `url.parse` with `native-url` npm package
- **Fix**: `test.js` mongodb deprecation warnings
- Updated `mqemitter-mongodb` to version 6.0.1
- **Feat** Added `createRetainedStreamCombi` and `outgoingEnqueueCombi` methods
- Dropped nodejs 6 support from `.travis.yml`
- Added nodejs 12 support on `.travis.yml`
- **Docs**: Updated readme with new features

### CODE REVIEW

**Comment by robertsLando** [comment]
@mcollina  I have also made the support for `createRetainedStreamCombi` and `outgoingEnqueueCombi`

**Comment by robertsLando** [comment]
Dropping nodejs 6 support in travis as also `aedes-persistence-redis` did it. Tests are good now

**Comment by robertsLando** [comment]
@mcollina I have tried to upgrade aedes-persistenct to 8.0.0 but there are some tests that fails. Seems all tests are related to the property `messageId` that is` null` instead of `undefined`. I don't know if this should be fixed in aedes-persistence tests `abstract.js` or in this library. Let me know if I need to change something.

**Comment by mcollina** [comment]
The behavior has changed, and the behavior of this library would have to be fixed.

**Comment by robertsLando** [comment]
Ok if you explain me what has changed I could try to fix it in another pr. What about this one? Can you make a review? 

Daniel

File: persistence.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict'
 
 var util = require('util')
-var urlModule = require('url')
+var urlModule = require('native-url')
```
Comment (mcollina) [inline persistence.js]: Why this change?

File: persistence.js [hunk 2]
```diff
@@ -51,7 +60,7 @@ MongoPersistence.prototype._setup = function () {
 
   var that = this
 
-  this._connect(function (err, client) {
+  this._connect(async function (err, client) {
```
Comment (mcollina) [inline persistence.js]: This should not be an async function, `_connect` is not handling it.

**Comment by mcollina** [comment]
See https://github.com/mcollina/aedes-persistence/pull/31, https://github.com/mcollina/aedes-persistence/pull/33 and https://github.com/mcollina/aedes-persistence/pull/32

File: persistence.js (hunk 1 above)
Comment (robertsLando) [reply persistence.js]: > 'url.parse' was deprecated since v11.0.0. Use 'url.URL' constructor instead node/no-deprecated-api

Unfortunally `new Url` doesn't work much like `url.parse` so this is a lightweight replacement for it (https://github.com/GoogleChromeLabs/native-url#readme)

File: persistence.js (hunk 2 above)
Comment (robertsLando) [reply persistence.js]: Done

**Comment by robertsLando** [comment]
@mcollina Just one question: Why [this](https://github.com/mcollina/aedes-persistence-mongodb/blob/master/test.js#L479) very long timeout in the test?

File: persistence.js [hunk 3]
```diff
@@ -32,15 +41,14 @@ function MongoPersistence (opts) {
 
 util.inherits(MongoPersistence, CachedPersistence)
 
-MongoPersistence.prototype._connect = function (cb) {
+MongoPersistence.prototype._connect = async function (cb) {
   if (this._opts.db) {
-    cb(null, this._opts.db)
+    await cb(null, this._opts.db)
```
Comment (mcollina) [inline persistence.js]: Please remove async/await.

Comment (robertsLando) [reply persistence.js]: Can I use promises instead?

Comment (mcollina) [reply persistence.js]: Unfortunately no as most of the other aedes codes does not use them and it would create a weird error model.

**Comment by robertsLando** [comment]
I think it's ok now. Could you take a look to #27 please? It is a critical bug and I think it's the root cause of all my problems with aedes right now.

**Comment by robertsLando** [comment]
@mcollina Travis build fails for no reason, I try to trigger new builds hoping it fixes, seems stucked

**Comment by robertsLando** [comment]
@mcollina should be ok now

**Comment by mcollina** [comment]
I think dropping indexes at every start is going to be extremely disruptive for long-running operations. Can you keep that behavior behind an option?

**Comment by robertsLando** [comment]
@mcollina Yout right, fixed

**Comment by robertsLando** [comment]
@mcollina Done. There is one test that fails sometimes, it check two dates and the difference is of 1ms, I have no idea why

**Comment by robertsLando** [comment]
@mcollina Should be ok now
